### DISCUSSION TITLE
共享农场第6期优化内容说明

### DISCUSSION THREAD

**Discussion by otto-001** [famland/website#190] [通告]
### **一、关于Fanland耕种收益机制优化说明**

**1、降低合约资源消耗，将APY定期自动计算机制调整为用户实时计算机制。**

现阶段的APY计算机制，是由核心合约每5分钟计算一次各农场APY数值后，供所有Famland计算各自的收益数额。此机制在试运营期间，每日须耗费的1-3EOS的资源，来计算APY。此机制不利于YLE生态持续性发展。

本着谁贡献，谁收获的设计原则，Famland的收益APY的计算调整为用户行为实时计算APY收益的机制。也就是当用户的农田进出农场、新增做市资金、收取收益等操作时，将实时计算各田的收益时长，根据当前农场的分发池余额、农场做市总资金量实时计算出农场的年化APY。

通过该机制的优化后，可以停止合约在定时计算APY方面资源的消耗。

Famland新的年化APY计算规则：

APY=所在农场分发池余额*30%/(N*T)

N：农场总做市资金效能，当用户Famaland进出农场、增加做市资金时更新该参数。N的数值按照农场内所有Famland做市本金中USDT数额*2来确定，不会因为YL价格波动而变化。

T：农场总收益时长，单位为秒，用户在每次进行Rebase操作，或进行主动或被动收取收益相关操作时，更新这个参数数值。每次收取收益将扣减累加的累计收益时长。每次Rebase操作时将累加到农场总的收益时长中。

**2、将收取收益和Rebase功能结合，大幅提升用户Rebase操作的积极性，为生态肥力值提升做贡献。**

目前YLE玩家由于主动Rebase的积极性不足，导致生态的肥力值增量未能达到最大值，从而影响生态总发行YL的最大流速。由此，为提升玩家Rebase的积极性，将Famland收取收益的机制和Rebase功能进行结合。玩家如果主动进行rebase操作，会更新提升名下各个Famalnd收益时长，增加各Famland实际收益额。

**3、按照做市时的实际本金计算收益，做市本金不会因为YL价值变化而变化。**

新的机制将按照Famland做市的实际额度来计算收益，而替换根据Famland做市的实时价值来计算收益。因为每个玩家通过Famland做市满额基本田，所作出的贡献大小价值是等同的，不会因为YL的价格出现波动，而影响玩家所作出贡献的价值。不论是之前、现在、还是将来，根据玩家实际做出贡献的大小计算相应的收益。

比如，玩家在Famalnd做市一块满额基本田，一共存入了250U+700YL（做市时，YL价格为0.35USDT），那么系统将按照500U的做市本金和200%的肥力加成以及实时APY来计算该田的收益。即便以后YL价格变为0.10USDT时，系统仍然按照500U的做市本和200%的肥力加成以及实时APY来计算该田的收益。

### **二、关于共享农场YL分配机制优化说明。**

根据共享农场第4期优化改进机制，YLE各农场根据其平均共享值的获得相应的新发行YL分配。但是该平均共享值的计算规则并未真实体现玩家实际贡献的实际价值。因此，需要在该需求基础上，对农场Famland的平均共享值计算规则进行优化，让农场玩家付出的贡献能获得对等的权益效应。

这里所指的贡献，包括玩家存入的资金、肥力值转化的加成效应、Famland数量规模效应、加成八宝树的运气效果都能在农场的的平均共享值中体现出来。

有做市过原始田的Famland共享值计算：

Famland的共享值= (当前实际共享值+1000) 

未做市过原始田的Famland共享值计算：

Famland的共享值=  当前实际共享值 

**农场的平均共享值的计算：**

共享农场Famland有效数量＜30时，

共享农场平均共享值=农场内所有Famland共享值*肥力加成系数之和 / 30

共享农场Famland有效数量≥30时

共享农场平均共享值=农场内所有Famland共享值*肥力加成系数之和 / 农场Famland有效数量

该机制执行后，将开放原始田做市通道，玩家可根据需要做市原始田。

### 三、关于开放原始田做市的说明

**原始田做市规则：**

1、玩家Famland肥力值达到600万时，或者玩家支付500USDT手续费，可以激活原始田做市入口，激活手续费中的50%用于项目开发，50%存入稳定池。

2、玩家可以在原始田做市后，原始田做市最多可存入500USDT+同等价值数额的YL。例如：玩家在原始田每存入1USDT+等值数量的YL可以获得2个共享值的提升。 

### 四、关于落实Famland肥力值及肥力加成系数转化的说明。

**Famland的肥力值计算规则：**

**肥力值=耕种时长×3%×（基本田和原始田做市存入的USDT数量 * 2 * 3%* 肥力加成系数+原始田、基本田野田做市存入的USDT数量*2之和×0.03%）**

**Famland肥力加成转化规则：**

Famland肥力值转化加成系数的前提，是需要在Famland上栽种八宝树CryptoTree NFT
Famland栽种八宝树CryptoTree NFT后，肥力加成可以按照0.001为单位进行提升，首次肥力加成提升0.001，肥力值须提升20000，肥力加成再次提升0.001，需提升的肥力值在之前的提升的基础值上提升3‰。

玩家可以根据以上规则手动将累计的肥力值转化成肥力加成系数。

**Comment by famland16816** [comment]
这个原始田要改成NFT，我想就叫做加速器或者说加速装置，因为它的作用就是加速。
我们来看一下他是怎样加速的。
首先它可以直接提升共享值。
就是你投入多少就可以提升多少，最高1000。
你可以直接提到1000。
而基本田要提升共享值需要，看你收取的收益，而这个收益是慢慢积累的，也就是说你要提升共享值，慢慢的提高，开放这个原始田，就可以加速这个共享值的提升。
这个共享值的加速提升又可以加速你共享农场的流速。
还有肥力值的加成，那么就等于加速了你的收益。

**Comment by famland16816** [comment]
原始田投入可以提升共享值，然后可以提升流速，但是他没有创造增发量。这个是否合理？

如果不考虑这个增发量，那么就没有做本质上的改变。之前有原始田的，不受农场欢迎，就是因为有原始田的会抢其他农田的收益，就是抢分发量。
那么现在要改成可以提升共享值，可以提升流速，那么本质还是抢分发量。

我们来看一下，假如这个原始田投入，能够带来增发量，会怎么样？

假如整个农场只有两块田，一号田只有基本田，500u。2号田基本田500 u，原始田1000u，那么按现有的机制，增发量只有693.5×2=1387
那么按照我的设想，原始田1000u也可以创造，693.5×2=1387增发量。

这样增发量就比原来多了一倍，相当于流速增加了一倍，收益也增加了一倍。

这个增发量就是创造的价值，是大家共享的，那么也就是说先富的人，他投入原始田，创造的价值，就相当于和后富的人共享了。
这个就是我们价值共享的理念。

**Comment by famland16816** [comment]
建议在这个农田的创建上做一个限制，比如说每个农户可创建的农田，最多只能三个。（这里有个前提就是技术上实现，一个人只能绑定一个账户）

这样想多种田的农场主。他三块田都种满了以后只能到市场上去买，或者租。
这样子做的好处。
一是可以激活农田nft，交易市场。
二是降低这个进入门槛。

对于没有资金的新人来说。他可以一块田出售，一块田出租，一块田留给自己种。
这个创建规则还是之前的，就是你这个田创建以后必须，出租或者出售出去或者做市以后，才可以创建下一块田。

其实这个还是一个先富帮后富，还是价值共享的理念，人人为我，我为人人！
